### ISSUE TITLE
Preload next track when approaching the end of the current one

### ISSUE THREAD

**Issue by plietar** [plietar/librespot#18]
Currently we don't start loading the next track until the current one has completed. This causes a short delay between tracks.

Instead the next track should start loading before that.

**Comment by fbrinker** [comment]
Maybe with (optional) crossfading, if possible

**Comment by TonioRoffo** [comment]
Crossfading and gapless playback are two different things.  In HW spotify connect gear, gapless is standard.

**Comment by RafaPolit** [comment]
Gapless is really a must for me, since most of what I listen to is either live or classical music, where much of the 'transitions' are mid-sound (specially in Opera, for instance).

Any time frame on this new feature? Thanks for all the hard work,  best regards,
Rafa.

**Comment by plietar** [comment]
Gapless isn't too hard, and I'm working on a refactor which will also enable that. I can't give it an ETA, but probably somewhere within the next month.

Cross-fade is a lot trickier, and I don't have any immediate plans for it

**Comment by cortegedusage** [comment]
+1 for gapless for me.
(Just for the record, no pressure, keep up the good work.)

**Comment by johnstok** [comment]
+1 – would be great if this was implemented.

**Comment by RafaPolit** [comment]
Any news on this? I'm eagerly awaiting for this and 6 months ago you said it wasn't hard.  Was it more difficult than expected or this is simply not a priority for others?  For me its critical.  Please?  Thanks.

**Comment by sashahilton00** [comment]
Paul mentioned he was busy until July, most things have been put on hold till then. I'm sure it will be added in due course. Cross fade is harder due to the need to mix audio streams

**Comment by RafaPolit** [comment]
Yeah, I wasn't talking about crossfading, just gapless playback for concerts, and classical pieces that flow seamlessly into the next movement (most critical in Opera).

**Comment by tomcek42** [comment]
Any progress on gapless playback? I still hear breaks between the tracks.

**Comment by ThatOneCalculator** [comment]
It's been about 2 years, any updates? I just switched to spotifyd and as much as it's good, I really really miss crossfade.

**Comment by wfraser** [comment]
@ThatOneCalculator: this isn't the main repository for librespot anymore, it's moved to https://github.com/librespot-org/librespot.

The feature in question has already been implemented there, in https://github.com/librespot-org/librespot/pull/430

**Comment by ThatOneCalculator** [comment]
Ty
